### ISSUE TITLE
Add functional `keras.ops.grad`

### ISSUE THREAD

**Issue by LarsKue** [keras-team/keras#22442]
It might prove beneficial to add a single, backend-agnostic version to get gradients with a simple functional `keras.ops.grad` or `keras.ops.value_and_grad` / `keras.ops.grad_and_value`.

Example:

```python
import keras

def f(_x):
    return 2.0 * _x

x = keras.ops.ones(())
grad_fn = keras.ops.value_and_grad(f, argnums=0, has_aux=False)
y, dfdx = grad_fn(x)

assert dfdx == 2.0
```

I **already have a working implementation** of this, so happy to open a PR should there be interest from the maintainers.

Some open questions:
1. Should we prefer the jax way (`value_and_grad`), the torch way (`grad_and_value`), or expose both? There is some slight overhead involved with swapping the return values.
2. Wrapping the inner function is done very neatly in jax, which adds extra information to the `__doc__` attribute of the wrapped function. What is the target here for keras?
3. Should this kind of function live in the `keras.ops` scope or would it be more sensible to introduce a `keras.autodiff` module or similar?
4. I have limited experience with tensorflow, so would be eager to hear if there are any other reservations for this kind of functionality, especially with respect to efficiency.

**Comment by hertschuh** [comment]
@LarsKue 

> I already have a working implementation of this, so happy to open a PR should there be interest from the maintainers.

For all backends?

While at first sight it seems like a great addition, I'm going to question the idea. How gradients work is _very_ different between JAX, TensorFlow, and Torch. While we may be able to create an API that appears to work on the surface level, users will inevitably run into issues where different backends behave differently, and we (Keras) won't be able to abstract those differences.

The high level API we offer and support on top of gradients is `model.fit()`, and there is a lot of code around `fit()` to make it work (and to make the gradients work the way we need them).

`model.fit()` also offers a lot of customizations so that you don't have to write your own custom training loop and manipulate gradients directly (outside of the optimizer):
- https://youtu.be/br15hZmKk2s?si=N29GRaed3tXkmoOZ&t=903
- https://youtu.be/br15hZmKk2s?si=UFBVq5D05XmqXJqj&t=958
- https://youtu.be/br15hZmKk2s?si=UFBVq5D05XmqXJqj&t=958

**Comment by LarsKue** [comment]
@hertschuh 

> For all backends?

Yes, you can view it here: [jax](https://github.com/bayesflow-org/bayesflow/blob/backend-autograd/bayesflow/backend/_jax/autodiff.py), [tensorflow](https://github.com/bayesflow-org/bayesflow/blob/backend-autograd/bayesflow/backend/_tensorflow/autodiff.py), [torch](https://github.com/bayesflow-org/bayesflow/blob/backend-autograd/bayesflow/backend/_torch/autodiff.py). Note I have not conducted thorough testing how this behaves on stateful functions, but my use-case is covered fine with this implementation.

Which challenges with respect to unifying backends for this API do you have in mind? I am happy to work on improving my initial simple implementation. The current one already side-steps potential issues by relying on available functional APIs from the respective backends where available.
